builtin/providers/aws/resource_aws_db_instance.go

```diff
@@ -184,64 +185,60 @@ func resourceAwsDbInstance() *schema.Resource {
 }
 
 func resourceAwsDbInstanceCreate(d *schema.ResourceData, meta interface{}) error {
-	conn := meta.(*AWSClient).rdsconn
-	opts := rds.CreateDBInstance{
-		AllocatedStorage:     d.Get("allocated_storage").(int),
-		SetAllocatedStorage:  true,
-		DBInstanceClass:      d.Get("instance_class").(string),
-		DBInstanceIdentifier: d.Get("identifier").(string),
-		DBName:               d.Get("name").(string),
-		MasterUsername:       d.Get("username").(string),
-		MasterUserPassword:   d.Get("password").(string),
-		Engine:               d.Get("engine").(string),
-		EngineVersion:        d.Get("engine_version").(string),
+	conn := meta.(*AWSClient).awsRDSconn
+	opts := rds.CreateDBInstanceMessage{
+		AllocatedStorage:     aws.Integer(d.Get("allocated_storage").(int)),
+		DBInstanceClass:      aws.String(d.Get("instance_class").(string)),
+		DBInstanceIdentifier: aws.String(d.Get("identifier").(string)),
+		DBName:               aws.String(d.Get("name").(string)),
+		MasterUsername:       aws.String(d.Get("username").(string)),
+		MasterUserPassword:   aws.String(d.Get("password").(string)),
+		Engine:               aws.String(d.Get("engine").(string)),
+		EngineVersion:        aws.String(d.Get("engine_version").(string)),
```

Comment:
Guys are you sure this is even a typed language we're using? Feels so easy and free! :neutral_face: